mount/upgrader.go

```diff
@@ -131,9 +131,13 @@ func (u *Upgrader) Close() error {
 }
 
 func (u *Upgrader) refetch(ctx context.Context) error {
-	_ = os.Remove(u.transientFilePath())
+	u.lk.Lock()
```

Comment:
@raulk 

1. If one refetch fails because of a transient issue , all contending go-routines that saw the same `sync.Once` will also fail to fetch the file. 

2. We don't clean up the temp file if fetching from the underlying Mount fails.

Reply:
> If one refetch fails because of a transient issue , all contending go-routines that saw the same sync.Once will also fail to fetch the file.

Yes, this is what we want.

Reply:
> We don't clean up the temp file if fetching from the underlying Mount fails.

Good point, absolutely need to do that cleanup.